app/stores/ada/AdaTransactionBuilderStore.js

```diff
@@ -208,9 +209,22 @@ export default class AdaTransactionBuilderStore extends Store {
   }
 
   @action
-  _reset = async () => {
+  _preWaitReset = () => {
+    // resets that have to happen BEFORE waiting for createUnsignedTx to finish
     this.plannedTxInfo = [{ address: undefined, value: undefined }];
     this.shouldSendAll = false;
+  }
+  @action
+  _postWaitReset = () => {
+    // resets that have to happen AFTER waiting for createUnsignedTx to finish
+    this.createUnsignedTx.reset();
+    this.plannedTx = null;
+    this.tentativeTx = null;
+  }
+
+  @action
+  _reset = async () => {
```

Comment:
Is _reset even used after the changes?

Reply:
Yes we have a listener on it that's called by the WalletSendPage

Reply:
(Heads up: I’m working on my iPad so it’s tricky to review the whole files)